### PR TITLE
docs(admin-api/license) Rework licenses admin api doc

### PR DESCRIPTION
**Pull request by lena-larionova**
Docs ticket: https://konghq.atlassian.net/browse/DOCS-1593

### Changes
* Changing double quotes (`" "`) around licenses to single quotes (`' '`)
* Link to deploy a license from license overview
* Cleaning up the Admin API reference and examples for clarity, formatting, flow

### Previews
[Licenses Admin API reference](https://deploy-preview-2607--kongdocs.netlify.app/enterprise/2.3.x/admin-api/licenses/reference/)
[Licenses examples](https://deploy-preview-2607--kongdocs.netlify.app/enterprise/2.3.x/admin-api/licenses/examples/)

### CODE REVIEW

**Comment by lena-larionova** [comment]
@mikefero Heather (@HCloward ) caught an issue when testing the Admin API instructions for applying a license - namely, the double quotes don't work. Fixing that here, and also had the chance now to go through the doc and clean it up a bit. Can you take a quick look?

File: app/enterprise/2.3.x/admin-api/licenses/examples.md [hunk 1]
```diff
@@ -71,10 +79,14 @@ http GET :8001/licenses/30b4edb7-0847-4f65-af90-efbed8b0161f
 
 ### Auto-generated ID
 
+To generate an ID automatically, submit a `POST` request directly to `/licenses`:
+
 ```bash
-http POST :8001/licenses payload="{\"license\":{\"payload\":{\"admin_seats\":\"1\",\"customer\":\"Example Company, Inc\",\"dataplanes\":\"1\",\"license_creation_date\":\"2017-07-20\",\"license_expiration_date\":\"2017-07-20\",\"license_key\":\"00141000017ODj3AAG_a1V41000004wT0OEAU\",\"product_subscription\":\"Konnect Enterprise\",\"support_plan\":\"None\"},\"signature\":\"6985968131533a967fcc721244a979948b1066967f1e9cd65dbd8eeabe060fc32d894a2945f5e4a03c1cd2198c74e058ac63d28b045c2f1fcec95877bd790e1b\",\"version\":\"1\"}}"
+$ http POST :8001/licenses \
+  payload='{\"license\":{\"payload\":{\"admin_seats\":\"1\",\"customer\":\"Example Company, Inc\",\"dataplanes\":\"1\",\"license_creation_date\":\"2017-07-20\",\"license_expiration_date\":\"2017-07-20\",\"license_key\":\"00141000017ODj3AAG_a1V41000004wT0OEAU\",\"product_subscription\":\"Konnect Enterprise\",\"support_plan\":\"None\"},\"signature\":\"6985968131533a967fcc721244a979948b1066967f1e9cd65dbd8eeabe060fc32d894a2945f5e4a03c1cd2198c74e058ac63d28b045c2f1fcec95877bd790e1b\",\"version\":\"1\"}}'
```
Comment (mikefero) [inline app/enterprise/2.3.x/admin-api/licenses/examples.md:86]: @lena-larionova The change doesn't appear to work locally for me; however the original one does:

#### Original
```
[kong-ee-dc0112767:/kong]# http POST :8001/licenses payload="{\"license\":{\"payload\":{\"admin_seats\":\"1\",\"customer\":\"Example Company, Inc\",\"dataplanes\":\"1\",\"license_creation_date\":\"2017-07-20\",\"license_expiration_date\":\"2017-07-20\",\"license_key\":\"00141000017ODj3AAG_a1V41000004wT0OEAU\",\"product_subscription\":\"Konnect Enterprise\",\"support_plan\":\"None\"},\"signature\":\"6985968131533a967fcc721244a979948b1066967f1e9cd65dbd8eeabe060fc32d894a2945f5e4a03c1cd2198c74e058ac63d28b045c2f1fcec95877bd790e1b\",\"version\":\"1\"}}"
HTTP/1.1 201 Created
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Length: 597
Content-Type: application/json; charset=utf-8
Date: Mon, 22 Feb 2021 21:14:31 GMT
Server: kong/2.3.2.0-dev-enterprise-edition
X-Kong-Admin-Latency: 12
X-Kong-Admin-Request-ID: kr8iin0Y14QZTKUZoudmx4AJFFL67WMb
vary: Origin

{
    "created_at": 1614028471,
    "id": "c1d0eb24-2f78-4e45-ad36-f88aac73496a",
    "payload": "{\"license\":{\"payload\":{\"admin_seats\":\"1\",\"customer\":\"Example Company, Inc\",\"dataplanes\":\"1\",\"license_creation_date\":\"2017-07-20\",\"license_expiration_date\":\"2017-07-20\",\"license_key\":\"00141000017ODj3AAG_a1V41000004wT0OEAU\",\"product_subscription\":\"Konnect Enterprise\",\"support_plan\":\"None\"},\"signature\":\"6985968131533a967fcc721244a979948b1066967f1e9cd65dbd8eeabe060fc32d894a2945f5e4a03c1cd2198c74e058ac63d28b045c2f1fcec95877bd790e1b\",\"version\":\"1\"}}",
    "updated_at": 1614028471
}
```

#### Changed; quotes removed

```
[kong-ee-dc0112767:/kong]# http POST :8001/licenses \
>   payload='{\"license\":{\"payload\":{\"admin_seats\":\"1\",\"customer\":\"Example Company, Inc\",\"dataplanes\":\"1\",\"license_creation_date\":\"2017-07-20\",\"license_expiration_date\":\"2017-07-20\",\"license_key\":\"00141000017ODj3AAG_a1V41000004wT0OEAU\",\"product_subscription\":\"Konnect Enterprise\",\"support_plan\":\"None\"},\"signature\":\"6985968131533a967fcc721244a979948b1066967f1e9cd65dbd8eeabe060fc32d894a2945f5e4a03c1cd2198c74e058ac63d28b045c2f1fcec95877bd790e1b\",\"version\":\"1\"}}'
HTTP/1.1 400 Bad Request
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Length: 215
Content-Type: application/json; charset=utf-8
Date: Mon, 22 Feb 2021 21:14:41 GMT
Server: kong/2.3.2.0-dev-enterprise-edition
X-Kong-Admin-Latency: 3
X-Kong-Admin-Request-ID: 39sC7lcMEna7fqz5NnriAogGWZi6BG57
vary: Origin

{
    "code": 2,
    "fields": {
        "payload": "Unable to validate license: could not decode license json"
    },
    "message": "schema violation (payload: Unable to validate license: could not decode license json)",
    "name": "schema violation"
}
```

If you want to remove the starting quotes the escaping of the double quote will also need to be removed:

```
http :8001/licenses payload='{"license":{"payload":{"admin_seats":"1","customer":"Example Company, Inc","dataplanes":"1","license_creation_date":"2017-07-20","license_expiration_date":"2017-07-20","license_key":"00141000017ODj3AAG_a1V41000004wT0OEAU","product_subscription":"Konnect Enterprise","support_plan":"None"},"signature":"6985968131533a967fcc721244a979948b1066967f1e9cd65dbd8eeabe060fc32d894a2945f5e4a03c1cd2198c74e058ac63d28b045c2f1fcec95877bd790e1b","version":"1"}}'
```

File: app/enterprise/2.3.x/admin-api/licenses/examples.md [hunk 2]
```diff
@@ -86,10 +98,15 @@ http POST :8001/licenses payload="{\"license\":{\"payload\":{\"admin_seats\":\"1
...
+`/licenses/<uuid>`:
+
 ```bash
-http PUT :8001/licenses/e8201120-4ee3-43ca-9e92-3fed08b1a15d payload="{\"license\":{\"payload\":{\"admin_seats\":\"1\",\"customer\":\"Example Company, Inc\",\"dataplanes\":\"1\",\"license_creation_date\":\"2017-07-20\",\"license_expiration_date\":\"2017-07-20\",\"license_key\":\"00141000017ODj3AAG_a1V41000004wT0OEAU\",\"product_subscription\":\"Konnect Enterprise\",\"support_plan\":\"None\"},\"signature\":\"6985968131533a967fcc721244a979948b1066967f1e9cd65dbd8eeabe060fc32d894a2945f5e4a03c1cd2198c74e058ac63d28b045c2f1fcec95877bd790e1b\",\"version\":\"1\"}}"
+$ http PUT :8001/licenses/e8201120-4ee3-43ca-9e92-3fed08b1a15d \
+  payload='{\"license\":{\"payload\":{\"admin_seats\":\"1\",\"customer\":\"Example Company, Inc\",\"dataplanes\":\"1\",\"license_creation_date\":\"2017-07-20\",\"license_expiration_date\":\"2017-07-20\",\"license_key\":\"00141000017ODj3AAG_a1V41000004wT0OEAU\",\"product_subscription\":\"Konnect Enterprise\",\"support_plan\":\"None\"},\"signature\":\"6985968131533a967fcc721244a979948b1066967f1e9cd65dbd8eeabe060fc32d894a2945f5e4a03c1cd2198c74e058ac63d28b045c2f1fcec95877bd790e1b\",\"version\":\"1\"}}'
```
Comment (mikefero) [inline app/enterprise/2.3.x/admin-api/licenses/examples.md:106]: Same for all these changes.

File: app/enterprise/2.3.x/admin-api/licenses/examples.md (hunk 1 above)
Comment (lena-larionova) [reply app/enterprise/2.3.x/admin-api/licenses/examples.md:86]: And this is why I wanted to come back and test it... eek. Thanks!

Comment (lena-larionova) [reply app/enterprise/2.3.x/admin-api/licenses/examples.md:86]: Ideally, the less work for the user, the better - so if they don't have to escape every double quote, this is definitely the preferable option.

**Comment by lena-larionova** [comment]
@mikefero should be good now. Good reminder to remember to come back and test things... I've tested now, it works, and doesn't require users to escape every `"`, which is nice.

File: app/enterprise/2.3.x/admin-api/licenses/reference.md [hunk 3]
```diff
@@ -5,22 +5,22 @@ book: licenses
...
     Attributes | Description
     ---:| ---
-    `id` | The **License's** unique ID.
+    `id` | The **license's** unique ID.
 
 licenses_body: |
     Attribute | Description
     ---:| ---
-    `payload` | The **{{site.ee_product_name}} License** in JSON format.
+    `payload` | The **Kong Gateway license** in JSON format.
```
Comment (mikefero) [inline app/enterprise/2.3.x/admin-api/licenses/reference.md:13]: Should this be `{{site.ee_base_gateway}}` or `{{site.base_gateway}}`?

File: app/enterprise/2.3.x/admin-api/licenses/reference.md [hunk 4]
```diff
@@ -5,22 +5,22 @@ book: licenses
...
 
 ## Introduction
 
-The {{site.ee_product_name}} Licenses feature is configurable through the
-[Admin API]. This feature allows the configuration a license in your
-{{site.ee_product_name}} cluster; traditional and hybrid mode deployments. In
-hybrid mode deployments, licenses configured via the `/licenses` endpoint will
-be sent to all dataplanes in the cluster and the most recent `updated_at`
-license will be used.
+The {{site.ee_base_gateway}} Licenses feature is configurable through the
```
Comment (mikefero) [inline app/enterprise/2.3.x/admin-api/licenses/reference.md:18]: I just looked at the rendered documentation and `{{site.ee_base_gateway}}` is blank; is this something that coming in a different branch of the docs?

**Review by mikefero — COMMENTED** [review]
Your changes look so much better than my throwing darts on a page; nice work.

I have two more questions.

File: app/enterprise/2.3.x/admin-api/licenses/reference.md (hunk 3 above)
Comment (lena-larionova) [reply app/enterprise/2.3.x/admin-api/licenses/reference.md:13]: It breaks when you try to use it in yaml front matter, so unfortunately it has to be plaintext here. Like so: https://docs.konghq.com/enterprise/2.3.x/admin-api/licenses/reference/#request-body

File: app/enterprise/2.3.x/admin-api/licenses/reference.md (hunk 4 above)
Comment (lena-larionova) [reply app/enterprise/2.3.x/admin-api/licenses/reference.md:18]: Ack - no, that's a typo, should just be {{site.base_gateway}}

**Review by mikefero — APPROVED** [review]
Thanks again for the cleanup

File: app/enterprise/2.3.x/admin-api/licenses/reference.md [hunk 5]
```diff
@@ -97,15 +98,17 @@ HTTP 201 Created
 
 {{ page.licenses_attribute_id }}
 
-* The behavior of `PUT` endpoint is the following: if the request payload
+When using `PUT`, if the request payload
 **does not** contain an entity's primary key (`id` for Licenses), the
```
Comment (HCloward) [inline app/enterprise/2.3.x/admin-api/licenses/reference.md:102]:
```suggestion
**does not** contain an entity's primary key (`id` for licenses), the
```

File: app/enterprise/2.3.x/admin-api/licenses/reference.md [hunk 6]
```diff
@@ -132,14 +135,16 @@ HTTP 200 OK
 
 {{ page.licenses_attribute_id }}
 
-* The behavior of `PATCH` endpoint is the following: if the request payload
+When using `PATCH`, if the request payload
 **does** contain an entity's primary key (`id` for Licenses), the license will
```
Comment (HCloward) [inline app/enterprise/2.3.x/admin-api/licenses/reference.md:139]:
```suggestion
**does** contain an entity's primary key (`id` for licenses), the license will
```

File: app/enterprise/2.3.x/admin-api/licenses/reference.md [hunk 7]
```diff
@@ -132,14 +135,16 @@ HTTP 200 OK
...
-be replaced with the given payload attribute. If the request payload **does
+be replaced with the given payload attribute.
+
+If the request payload **does
 not** contain an entity's primary key (`id` for Licenses), a `404 NOT FOUND`
```
Comment (HCloward) [inline app/enterprise/2.3.x/admin-api/licenses/reference.md:143]:
```suggestion
not** contain an entity's primary key (`id` for licenses), a `404 NOT FOUND`
```

**Review by HCloward — APPROVED** [review]
I noticed you lowercased license in some places and not others. Not sure if it should be capitalized where I made some suggestions or not.

**Pull request merged**